Developer builds crash if Embedded Social API key isn't provided in gradle.properties

**Summary:** 

If you don't provide the `EmbeddedSocialApiKey_oba=xxxxxxxxxxxxx` line in your `gradle.properties` file (in the `/onebusaway-android` directory), when building, installing, and executing the app you'll see a crash:

~~~
 Process: com.joulespersecond.seattlebusbot, PID: 30919
    java.lang.RuntimeException: Unable to create service com.microsoft.embeddedsocial.service.WorkerService: java.lang.NullPointerException: Attempt to invoke virtual method 'com.microsoft.embeddedsocial.server.IAccountService com.microsoft.embeddedsocial.server.EmbeddedSocialServiceProvider.getAccountService()' on a null object reference
        at android.app.ActivityThread.handleCreateService(ActivityThread.java:3552)
        at android.app.ActivityThread.-wrap4(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1786)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6938)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'com.microsoft.embeddedsocial.server.IAccountService com.microsoft.embeddedsocial.server.EmbeddedSocialServiceProvider.getAccountService()' on a null object reference
        at com.microsoft.embeddedsocial.service.handler.SignInHandler.<init>(SignInHandler.java:43)
        at com.microsoft.embeddedsocial.service.IntentProcessorFactory.createIntentProcessor(IntentProcessorFactory.java:44)
        at com.microsoft.embeddedsocial.service.WorkerService.createIntentProcessor(WorkerService.java:23)
        at com.microsoft.embeddedsocial.base.service.AbstractProcessingService.onCreate(AbstractProcessingService.java:49)
        at android.app.ActivityThread.handleCreateService(ActivityThread.java:3542)
        at android.app.ActivityThread.-wrap4(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1786) 
        at android.os.Handler.dispatchMessage(Handler.java:105) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6938) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374) 
~~~

@acrown-msft I believe this is new behavior with v0.7.3.

**Steps to reproduce:** 

1. Make sure you don't have the `EmbeddedSocialApiKey_oba=xxxxxxxxxxx` line in `onebusaway-android/gradle.properties`
1. Clean and build and run the app

**Expected behavior:** 

Run without crashing (although Embedded Social features will be hidden)

**Observed behavior:** 

Crash with the above stack trace

**Device and Android version:** 

Samsung Galaxy S8+ with Android 8.0

--- a/onebusaway-android/src/main/java/org/onebusaway/android/util/EmbeddedSocialUtils.java
+++ b/onebusaway-android/src/main/java/org/onebusaway/android/util/EmbeddedSocialUtils.java
@@ -136,7 +136,8 @@ public class EmbeddedSocialUtils {
      * Returns true if the Embedded Social api key is non-empty
      */
     public static boolean isSocialApiKeyDefined() {
-        return !BuildConfig.EMBEDDED_SOCIAL_API_KEY.isEmpty();
+        String embeddedSocialApiKey = BuildConfig.EMBEDDED_SOCIAL_API_KEY;
+        return embeddedSocialApiKey != null && !embeddedSocialApiKey.isEmpty();
     }
 
     /**

